### DISCUSSION TITLE
ffmpeg-git version on archlinux causing troubles on updates

### DISCUSSION THREAD

**Discussion by Parakovsky** [savonet/liquidsoap#3596] [Installation]
![image](https://github.com/savonet/liquidsoap/assets/38038134/7b22abb5-3b22-4eb4-984b-2650ed117050)

felas, i got this error every time it asking me to update packages. since for my other needs i would prefer to keep the git version over default pacman version I would like to know if that is anything I should worry about. thank you.

**Comment by toots** [comment]
Hi @Parakovsky ! `opam` tries to do its best to use system packages for dependencies like these. It works for most users. If you are installing a custom ffmpeg, you can consider that you know what you are doing. In this case, `opam` will give you a warning that you can, indeed, ignore.

**Comment by Parakovsky** [comment]
Dear @toots I know you are working hard to keep this amazing software as sofisticated as possible, but you probably misunderstood me - which is no problem as I myself misunderstand myself all the time. Unfortunately, there are no way I know around that problem but to revert back to default ffmpeg packaging, which are not huge pain in you know where, but I would like to know one day where to go to fix this to make it work with a git version. Have an excellent day my friend.

**Reply by vitoyucepi** [reply]
Hi @Parakovsky,

I can recommend you to ask the same question in the https://github.com/ocaml/opam discussions, probably they are already aware of this kind of problem and can give you a more consistent advice.

As for the package, if I were you, I would try to patch and pin the [conf-ffmpeg](https://github.com/ocaml/opam-repository/tree/master/packages/conf-ffmpeg/conf-ffmpeg.1) package.
```diff
--- a/opam
+++ b/opam
@@ -11,8 +11,9 @@ depends: [
 depexts: [
   ["libavutil-dev" "libavformat-dev" "libavcodec-dev" "libavdevice-dev" "libavfilter-dev" "libswresample-dev" "libswscale-dev"] {os-family = "debian" | os-family = "ubuntu"}
   ["ffmpeg-dev"] {os-distribution = "alpine"}
-  ["ffmpeg"] {os = "freebsd" | os-distribution = "arch" | os-distribution = "nixos" | os = "macos" & os-distribution = "homebrew"}
+  ["ffmpeg"] {os = "freebsd" | os-distribution = "nixos" | os = "macos" & os-distribution = "homebrew"}
   ["ffmpeg-devel"] {os-distribution = "centos" | os-distribution = "fedora" | os-family = "suse" | os-family = "opensuse"}
+  ["ffmpeg-git"] {os-distribution = "arch"}
 ]
 synopsis: "Virtual package relying on FFmpeg"
 description:
```
Please have a look at the opam packaging documentation at https://opam.ocaml.org/doc/Packaging.html.

**Reply by vitoyucepi** [reply]
Alternatively, you can patch archlinux packages. For example, you could rename the ffmpeg-git package to ffmpeg, prevent the package from being updated, and maintain it manually.
Or you could make a copy of the ffmpeg package repository and build your own package by putting the desired git commit into it.

**Comment by Parakovsky** [comment]
Thanks this looks like something I would need, cheers!
